docs/ProviderDesign.md

```diff
@@ -0,0 +1,79 @@
+# Provider's view of the FLOCX
+
+![ProviderDesign](../images/ProviderService-Marketplace.png)
+
+
+
+
+## Multi-tenant IRONIC:
+Ironic manages multiple pools of hardware each owned by a separate user/group/organization.
+
+
+## Provider Service:
+A centralized service per OpenStack deployment that manages offers and contracts 
+on behalf of all its tenants of the OpenStack.
+
+## Keystone:
+Provides authentication and authorization services.
+
+
+## Workflow:
+
+Following steps describe the process of offering physical hardware in the market by individual owners
+and receiving contracts from the marketplace.
+
+
+
+1. Hardware pool owners decide which nodes to offer in the marketplace and submit offers to the provider service
+2. Provider service confirms owner-hardware relationship before accepting offers. 
+3. For each accepted offer it fetches the hardware configuration from Ironic
+4. These offers are published to the marketplace as soon as possible. 
+5. Receives contracts from the marketplace.
+
+
+## Offers (Owners --> Providers):
+An offer made by an owner and submitted to the provider service will have following attributes.
+* Node ID: Unique ID of the node
+* Start time: Date and time of availability
+* Duration: Hours, days that the node will be available
+* Floor Price: Minimum Price that is demanded.
+
+Provider validates the offer received from the owner  from keystone and adds the following attributes:
+* Offer ID: Unique ID for each offer.
+* Created_at:
+* Status: Waiting 
+
+    * Possible values: 
+    * Waiting: When provider accepts offer from the owner
+    * Published: When provider publishes the offer in the marketplace
+    * InContract: When an offer finds a suitable bid and a contract is formed.
+    * Expired: For any reason if the offer is cancelled or gets past its time of availability or gets used.
+
+Provider also fetches the node configuration from Ironic and adds it to the offer. 
+
+* Node Configuration: (as extracted from Ironic)
+
+## Offers (Providers --> Marketplace):
```

Comment:
I don't think we need this section, as marketplace offers are fully described in the marketplace doc. It may be enough to just mention that a provider's offer will be sent to the marketplace, alongside any ironic info.

Reply:
This was put here just to show the additional attributes that a published offer contains as compared to the one that is just submitted to the provider service and not yet published.

Reply:
I think it's better to link to the marketplace documentation; duplicating information in two different places is a good way for them to eventually go out of sync.

Reply:
understood. will link it.